guvnor-ala/guvnor-ala-openshift/guvnor-ala-openshift-provider/src/main/java/org/guvnor/ala/openshift/access/impl/OpenShiftAccessInterfaceImpl.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.ala.openshift.access.impl;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.utils.URLUtils;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftConfig;
+import io.fabric8.openshift.client.OpenShiftConfigBuilder;
+import okhttp3.TlsVersion;
+import org.guvnor.ala.openshift.access.OpenShiftAccessInterface;
+import org.guvnor.ala.openshift.access.OpenShiftClient;
+import org.guvnor.ala.openshift.config.OpenShiftProviderConfig;
+import org.guvnor.ala.openshift.model.OpenShiftProvider;
+import org.guvnor.ala.runtime.providers.ProviderId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.commons.lifecycle.Disposable;
+
+public class OpenShiftAccessInterfaceImpl implements OpenShiftAccessInterface, Disposable {
+
+    protected static final Logger LOG = LoggerFactory.getLogger(OpenShiftAccessInterfaceImpl.class);
+    private final Map<String, OpenShiftClient> clientMap = new ConcurrentHashMap<>();
+
+    @Override
+    public OpenShiftClient getOpenShiftClient(final ProviderId providerId) {
+        if (!clientMap.containsKey(providerId.getId())) {
+            clientMap.put(providerId.getId(), buildClient(providerId));
+        }
+        return clientMap.get(providerId.getId());
+    }
+
+    private OpenShiftClient buildClient(final ProviderId providerId) {
+        assert (providerId instanceof OpenShiftProvider);
```

Comment:
we don't usually use asserts, better check using `if` and throw exception if the case. we have a utils class to check instance of if it makes it easy :)

https://github.com/AppFormer/uberfire/blob/master/uberfire-commons/src/main/java/org/uberfire/commons/validation/Preconditions.java#L31